src/OmniSharp.Roslyn.CSharp/Services/SemanticHighlight/SemanticHighlightService.cs

```diff
@@ -0,0 +1,174 @@
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Text;
+using OmniSharp.Mef;
+using OmniSharp.Models.SemanticHighlight;
+
+namespace OmniSharp.Roslyn.CSharp.Services.SemanticHighlight
+{
+    [OmniSharpHandler(OmniSharpEndpoints.V2.Highlight, LanguageNames.CSharp)]
+    public class SemanticHighlightService : IRequestHandler<SemanticHighlightRequest, SemanticHighlightResponse>
```

Comment:
Just to clarify, this current impl doesn't take advantage of the partial semantic classification tech that's available right? 

Oh and "partial" in the sense of only updating a single lines classification if only that line changed.

I ask because for C# files this might be a huge performance hit.

Reply:
The highlight request has an optional range property so that only that TextSpan is classified.

Reply:
I thought VSCode re-requested the entire document pretty frequently though to account for external file changes, project changes etc. Maybe I'm wrong though, my information is coming from design discussions during the features creation; not out of personal trial and error.

Reply:
Ah, perhaps you are referring to the [`provideDocumentSemanticTokensEdits` api](https://github.com/microsoft/vscode/blob/d618f85bee6cfa2ce1b8eb50ca369dca58664e89/src/vs/vscode.proposed.d.ts#L292-L345)

Reply:
Yup exactly!

Reply:
After thinking about that API for a bit, I'm not sure the Edits api is super useful. It seems mostly to benefit requesting changes from a remote LSP where bandwidth is a concern. It also requires the LSP to hold on to previous state in order to generate the Edits diff as well as fully classify the new state of the document.

Reply:
I'm not entirely sure how the client interprets all of the data in a response but I imagine it'd also optimize the clients processing of the data

Reply:
Yeah, I feel like that api would basically be faked by the editor assuming they have a good method of generating the diffs from two sets of SyntaxTokens. Since they already have the before state which the LSP really shouldn't be expected to hold on to.